kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/util/Metrics.java

```diff
@@ -54,6 +55,8 @@ public class Metrics {
     private static final String PROXY_TO_SERVER_ERROR_BASE_METER_NAME = "kroxylicious_proxy_to_server_errors";
     private static final String CLIENT_TO_PROXY_CONNECTION_BASE_METER_NAME = "kroxylicious_client_to_proxy_connections";
     private static final String PROXY_TO_SERVER_CONNECTION_BASE_METER_NAME = "kroxylicious_proxy_to_server_connections";
+    private static final String PROXY_TO_SERVER_CONNECTION_BLOCKED_METER_NAME = "kroxylicious_proxy_to_server_connection_blocked";
+    private static final String CLIENT_TO_PROXY_CONNECTION_BLOCKED_METER_NAME = "kroxylicious_client_to_proxy_connection_blocked";
```

Comment:
"blocked" usually means "a _thread_ is blocking on something, such as I/O", but that's not what we mean here, so this name would be misleading. 

Really this is about TCP backpressure. We might say "the connection is muted", or "reading is paused", or "reading is throttled to 0", or similar. Can we pick a better term to use in the name (and later on the description)?

Another aspect about whatever we decide for the name... TCP is a duplex thing, reading might be paused by writing might not be. So the name should make clear which direction is affected, rather than just "connection".

Reply:
> So its consistent with our internal usage. 

That's not especially relevant because the audience for this is end users, not ourselves.

> The thread is blocked 

No it's not, because this is non-blocking IO.

Reply:
> > So its consistent with our internal usage. 
> 
> 
> 
> That's not especially relevant because the audience for this is end users, not ourselves.
We should be using terms consistently both within the code and externally.
> 
> 
> 
> > The thread is blocked 
> 
> 
> 
> No it's not, because this is non-blocking IO. 

IO not the only source of blocking. As far as users are concerned their connections cannot make progress, it is no interest to them if the thread is blocked in your very narrow definition of blocked or my wider unable to make progress version. The fact we haven't blocked a netty thread is in an implementation detail. The term would still be accurate and appropriate if we switched to using look.

Reply:
I'm more with Tom on this.  Blocked itself sufficiently precise.  Also isn't the directionality of ""kroxylicious_proxy_to_server_connection_blocked" the wrong way about:

I would go with:

"kroxylicious_proxy_to_server_connection_blocked" -> "kroxylicious_server_to_proxy_reads_paused"
"kroxylicious_client_to_proxy_connection_blocked" -> "kroxylicious_client_to_proxy_reads_paused"